librz/bin/format/elf/elf_notes.c

```diff
@@ -75,22 +72,24 @@ static bool parse_note_file(RzBinElfNote *note, Elf_(Nhdr) * note_segment_header
 	ut64 entry_offset = offset;
 	for (Elf_(Addr) i = 0; i < n_maps; i++) {
 		if (strings_offset + strings_array_len >= note_segment_header->n_descsz) {
-			break;
+			rz_vector_fini(&files);
+			return false;
 		}
 
-		char tmp[ELF_STRING_LENGTH] = { 0 };
 		ut64 string_offset = offset + strings_offset + strings_array_len;
-		if (rz_buf_read_at(bin->b, string_offset, (ut8 *)tmp, ELF_STRING_LENGTH - 1) < 0) {
-			break;
+		char *name = rz_buf_get_string(bin->b, string_offset);
```

Comment:
This might be risky as it is now... Potentially we could end up reading the whole binary into memory if there is no NULL byte in the binary.

Reply:
Can we limit the reading with the section size instead?

Reply:
I think section size is wrong. There may be edge cases where things cross the section/segments I think, as long as they are consecutive in the virtual space.

Reply:
There is no way to limit the string size. But there is a check at the beginning of the loop.

Reply:
Oh, actually this is ELF notes... so we could in theory limit it with the size of the section, I think.

Reply:
There is already the check at the beginning of the loop. What i was trying to say is that there isn't any hard cap or defined string size.

Reply:
The check at the beginning of the loop does not prevent to read the whole bin.. It's not a huge deal, but I think we should have a `rz_buf_get_nstring(RzBuffer *b, ut64 addr, size_t max_size)`, where we would call this with max_size=note_section->size`

Reply:
There isn't any function rz_buf_get_nstring right now. Should we wait, merge the PR or add the function on this PR

Reply:
As you prefer ;) Your call. As said, it's probably not a very big deal so far, so choose what you think is better.

Reply:
I add the function and update elf_notes.c